examples/raspberrypi/rp2xxx/src/dma.zig

```diff
@@ -36,9 +35,21 @@ pub fn main() !void {
     rp2xxx.uart.init_logger(uart);
 
     const channel = dma.claim_unused_channel();
-    channel.?.trigger_transfer(@intFromPtr(dst[0..dst.len].ptr), @intFromPtr(hello[0..hello.len].ptr), hello.len, .{ .enable = true, .read_increment = true, .write_increment = true, .data_size = .size_8, .dreq = .permanent });
-
-    while (channel.?.is_busy()) {}
+    
+    channel.?.trigger_transfer(
+        @intFromPtr(dst[0..dst.len].ptr),
```

Comment:
could we maybe make `trigger_transfer` take `*anyopaque`, and cast it to an int in the function? It would be more ergonomic for clients.

Reply:
Could you provide an example? While this is RP2040-specific, anyopaque seems like it could be useful for a more abstract DMA driver design

Reply:
`@compileError` over `@panic` though

Reply:
I would leave that for the future. It requires further consideration, consistency across other APIs, and likely shared code to handle pointers, slices, and integers when the underlying type is an address.